semana4/lista-de-exercicios/lista.js

```diff
@@ -0,0 +1,438 @@
+ guarda isso na constante cotacao. Depois retorna a string "R$" mais o valorEmDolar (argumento )
+ vezes a cotacao (constante).
+ Então, uma constante meuDinheiro é criada pra guardar a chamada da função com o argumento 
+ valorEmDolar de 100.
+ Ao chamar console.log(meuDinheiro), imprimirá no console o valor em reais de 100 dólares.
+ No caso da cotaçao fornecida ser igual a 5, 
+ apareceria no console "R$500"
+ */
+
+ /*Exercicio2
+ Temos uma função investeDinheiro, com os argumentos tipoDeInvestimento e valor. 
+ Na funçao, usamos a condicional switch case pra dizer o que será feito com cada valor de argumento tipoDeInvestimento. 
+ Por exemplo, no caso do argumento escolhido pelo usuário ser poupança, a variável valorAposInvestimento será 
+ o parametro valor dado pelo usuário vezes 1.03.
+ No caso do tipoDeInvestimento escolhido pelo usuário náo se encaixar em nenhum case, ele cairá no default e um alert surgirá
+ dizendo que o tipo de investimento informado está incorreto. 
+ Ao fim da função, retornará o valorAposInvestimento.
+ Fora do escopo da função, duas variáveis const : novoMontante, que chama a função e passa os argumentos como "Ações"(tipoDeInvestimento) e 150 (valor);
+ A outra variável, segundoMontante chama a função e passa os argumentos como "Tesouro Direto" e 200
+ Ao executar console.log(novoMontante), aparecerá no console 165.
+ Ao executar console.log(segundoMontante), uma caixa de alerta será mostrada com a mensagem "TIPO DE INVESTIMENTO INFORMADO INCORRETO!"
+ */
+
+ /* Exercício3
+ Temos três variáveis, um array com números (const numeros) e 
+ dois arrays ainda vazios (const array1 e const array2).
+ Depois, usando um for of, nós vamos "correr" pelo array numeros e selecionar 
+ os números contidos nesse array em pares e ímpares. 
+ Se eles forem pares (dividindo o número por 2 o resto é zero), 
+ ele será adicionado ao array1 (usando o array1.push(numero));
+ Senão (else), ele será adicionado ao array2 (usando o .push()).
+ Depois daremos um console.log em que pediremos o número total de números e, para isso,
+ utilizaremos o numeros.length. Será impresso no console : 
+ "Quantidade total de números 14".
+ Depois, utilizaremos o .length para fazer o mesmo com array1 e array2.
+ No console.log(array1.length), teremos no console o número 6;
+ No console.log(array2.length), teremos no console o número 8.
+*/
+
+/*Exercício4
+Nós temos um array com numeros e depois duas variaveis numero1, numero2. 
+Utilizando um for of, vamos percorrer todo o array e, no primeiro if, vai encontrar o menor número 
+existente na array numeros.
+no segundo if, vamos encontrar o maior numero existente na array numeros.
+No final, chamaremos um console.log(numero1), que imprimirá no console o número -10 e no
+console.log(numero2), imprimirá no console o número 1590.
+*/
+
+
+
+
+
+
+
+//LOGICA DE PROGRAMACAO
+
+/*Exercício1
+Três maneiras de se iterar um array é usando forEach, Map e filter.
```

Comment:
Na verdade era esperado que você respondesse for, for of e while. Mas gostei que você foi adiante!

Reply:
Só não esqueça que map e filter criam novos vetores!